brave/brave-core
# [Android] App menu fixes

App menu fixes:
1. Replace page info button with share button in app main_menu (used when bottom toolbar is not enabled.) and in it's footer (used when bottom toolbar is enabled).
2. Remove Share.. row in the app menu on phone and only show it on tablet.
3. Hide bookmark button in app menu footer when bottom toolbar is enabled

Resolves https://github.com/brave/brave-browser/issues/8584
Resolves https://github.com/brave/brave-browser/issues/8583

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
[Phone]
1. Open brave with bottom toolbar enabled
2. Open app menu through the three-dot button in the bottom toolbar
3. Check that there is no "share.." row in the app menu
4. Check that the bottom icon bar of the app menu has a share button and no info and bookmark button.
![Screenshot_20200309-091742](https://user-images.githubusercontent.com/4730197/76234368-e163ae00-61e6-11ea-8a64-1a3b24abb814.png)
5. Click on the share button and confirm it works as expected.
6. Disable bottom toolbar through Appearance settings
7. Open app menu through the three-dot button in the bottom toolbar
8. Check that there is no "share.." row in the app menu
9. Check the top icon bar of the app menu has a share and bookmark button but no info button.
![Screenshot_20200309-092009](https://user-images.githubusercontent.com/4730197/76234826-79fa2e00-61e7-11ea-8430-1070ae13e732.png)
10. Click on the share button and confirm it works as expected.

[Tablet only]
Check that "share.." row in the app menu would appear in normal webpages and would not appear in chrome pages.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
